hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/OverReplicatedProcessor.java

```diff
@@ -114,6 +114,9 @@ private void adjustPendingOps(ContainerID containerID, SCMCommand<?> cmd,
       DeleteContainerCommand rcc = (DeleteContainerCommand) cmd;
       pendingOps.scheduleDeleteReplica(containerID, targetDatanode,
           rcc.getReplicaIndex());
+      if (rcc.getReplicaIndex() > 0) {
```

Comment:
Inside the new RM code, we probably should have the `else rcc.getReplicaIndex() == 0`. Now, only EC containers will land here, but when we finally turn off the LegacyRM, Ratis containers will land here too. It would make sense to be ready for that. That said, there will be more changes needed to process Ratis containers in this code path, but not in this class I think.

Reply:
Initially, I tried adding only EC Metrics and left RATIS Metrics out of this as it doesn't follow this code path right now. But I can add it to make it future-proof.